ios/RNSScreenStackHeaderConfigComponentView.mm

```diff
@@ -0,0 +1,679 @@
+        //            navitem.hidesSearchBarWhenScrolling = searchBar.hideWhenScrolling;
+        //          }
+        //#endif
+        //        }
+        //      }
+        //      case RNSScreenStackHeaderSubviewTypeBackButton: {
+        //        break;
+        //      }
+    }
+  }
+
+  if (animated && vc.transitionCoordinator != nil &&
+      vc.transitionCoordinator.presentationStyle == UIModalPresentationNone && !wasHidden) {
+    // when there is an ongoing transition we may need to update navbar setting in animation block
+    // using animateAlongsideTransition. However, we only do that given the transition is not a modal
+    // transition (presentationStyle == UIModalPresentationNone) and that the bar was not previously
+    // hidden. This is because both for modal transitions and transitions from screen with hidden bar
+    // the transition animation block does not get triggered. This is ok, because with both of those
+    // types of transitions there is no "shared" navigation bar that needs to be updated in an animated
+    // way.
+    [vc.transitionCoordinator
+        animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext> _Nonnull context) {
+          [self setAnimatedConfig:vc withConfig:config];
+        }
+        completion:^(id<UIViewControllerTransitionCoordinatorContext> _Nonnull context) {
+          if ([context isCancelled]) {
+            UIViewController *fromVC = [context viewControllerForKey:UITransitionContextFromViewControllerKey];
+            RNSScreenStackHeaderConfigComponentView *config = nil;
+
+            // TODO: make sure this will exist in fabirc
+            for (UIView *subview in fromVC.view.reactSubviews) {
+              if ([subview isKindOfClass:[RNSScreenStackHeaderConfigComponentView class]]) {
+                config = (RNSScreenStackHeaderConfigComponentView *)subview;
+                break;
+              }
+            }
+            [self setAnimatedConfig:fromVC withConfig:config];
+          }
+        }];
+  } else {
+    [self setAnimatedConfig:vc withConfig:config];
+  }
+}
+
++ (void)addSubviewsToNavItem:(UINavigationItem *)navitem withConfig:(RNSScreenStackHeaderConfigComponentView *)config
+{
+  for (RNSScreenStackHeaderSubviewComponentView *subview in config.reactSubviews) {
+    switch (subview.type) {
+      case facebook::react::RNSScreenStackHeaderSubviewType::Left: {
+        UIBarButtonItem *buttonItem = [[UIBarButtonItem alloc] initWithCustomView:subview];
+        navitem.leftBarButtonItem = buttonItem;
+        break;
+      }
+      case facebook::react::RNSScreenStackHeaderSubviewType::Right: {
+        UIBarButtonItem *buttonItem = [[UIBarButtonItem alloc] initWithCustomView:subview];
+        navitem.rightBarButtonItem = buttonItem;
+        break;
+      }
+      case facebook::react::RNSScreenStackHeaderSubviewType::Center: {
+        navitem.titleView = subview;
+        break;
+      }
```

Comment:
This doesn't accommodate for an item (eg. logo) to be put in the center in place of the title. This could be added in newer versions tho

Reply:
I agree